src/dbnode/persist/fs/commitlog/commit_log.go

```diff
@@ -355,8 +409,15 @@ func (l *commitLog) write() {
 			l.pendingFlushFns = append(l.pendingFlushFns, write.callbackFn)
 		}
 
-		if now := l.nowFn(); !now.Before(l.writerState.writerExpireAt) {
-			err := l.openWriter(now)
+		var (
```

Comment:
super nit: make a helper `shouldRotateWrite` fn instead

Reply:
tried it, actually makes code messier. Gonna keep as is